Deleting nonexisting content in English (#3939)
mdn/translated-content: files/es/web/javascript/index.md

Esta sección está dedicada al lenguaje JavaScript en sí, y no a las partes que son específicas de las páginas web u otros entornos host. Para información acerca de {{Glossary("API","APIs")}} específicas para páginas Web, consulta [APIs Web](/es/docs/Web/API ) y {{Glossary("DOM")}}.

El estándar para JavaScript es [ECMAScript](/es/docs/Web/JavaScript/Language_Resources) (ECMA-262) y [la especificación de la API para la Internacionalización de ECMAScript](https://tc39.es/ecma402/) (ECMA-402). La documentación en MDN está basada enteramente en las últimas versiones preliminares de ECMA-262 y ECMA-402. Y en algunos casos donde algunas [propuestas para nuevas funciones para ECMAScript](https://github.com/tc39/proposals) ya hayan sido implementadas en los navegadores, la documentación y algunos artículos de MDN pueden hacer uso de algunas de estas funciones.

> No confundas JavaScript con el [lenguaje de programación Java](https://en.wikipedia.org/wiki/Java_(programming_language) ). Ambos "Java" y "JavaScript" son marcas o marcas registradas de Oracle en los Estados Unidos y otros países, Sin embargo, los dos lenguajes de programación tienen sintaxis, semántica y usos muy diferentes.

Reviewer: Como nos sugería cw118, de ser posible, estaria bueno cambiar los enlaces a referencias externas en Español (equivalente a la referencia en Inglés) 

```suggestion
No confundas JavaScript con el [lenguaje de programación Java](https://es.wikipedia.org/wiki/Java_(lenguaje_de_programaci%C3%B3n)). Ambos "Java" y "JavaScript" son marcas o marcas registradas de Oracle en los Estados Unidos y otros países, Sin embargo, los dos lenguajes de programación tienen sintaxis, semántica y usos muy diferentes.
```

Author: Listo, @JuanVqz ya lo modifique y le di una revisada por si quedaba otro link en ingles.

  - : Una descripción general para aquellos que _piensan_ que saben JavaScript.
- [Estructuras de datos en JavaScript](/es/docs/Web/JavaScript/Data_structures)
  - : Descripción general de las estructuras de datos disponibles en JavaScript.
- [Comparaciones de igualdad y similaridad](/es/docs/Web/JavaScript/Equality_comparisons_and_sameness)
  - : JavaScript proporciona tres diferentes operadores de comparación de valores: igualdad estricta usando `===`, igualdad abstracta usando `==`, y el método {{jsxref("Global_Objects/Object/is", "Object.is()")}}.
- [Cierres (clousures)](/es/docs/Web/JavaScript/Closures)
  - : Un cierre es la combinación de una función y un entorno léxico dentro del cual se declaró esa función.

### Avanzado

- [Herencia y la cadena de prototipos](/es/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
  - : Explicación de la herencia basada en prototipos ampliamente incomprendida y subestimada.
- [Modo estricto](/es/docs/Web/JavaScript/Reference/Strict_mode)
  - : El modo estricto define que no puedes usar ninguna variable antes de iniciarla. Es una variable restringida de ECMAScript 5, para un rendimiento más rápido y una depuración más fácil.

> - [Arreglos escritos en JavaScript](/es/docs/Web/JavaScript/Typed_arrays)

Reviewer 2: Verificar traducción `typed arrays` (`arreglos escritos`)

Author: Lo he sustituido por arreglos tipados.

  - : Los arreglos escritos en JavaScript proveen un mecanismo para acceder a datos binarios sin formato.
- [Gestión de la Memoria](/es/docs/Web/JavaScript/Memory_Management)
  - : Ciclo de vida de la memoria y recolección de basura en JavaScript.
- [Modelo de Concurrencia y Bucle de Eventos](/es/docs/Web/JavaScript/EventLoop)
  - : JavaScript tiene un modelo de concurrencia basado en un "bucle de eventos".

## Referencia

Consulta la documentación completa de la [referencia de JavaScript](/es/docs/Web/JavaScript/Reference).

> - [Objectos Estándar](/es/docs/Web/JavaScript/Reference/Global_Objects)

Reviewer 2: typo en "Objectos"

- [Expresiones y operadores](/es/docs/Web/JavaScript/Reference/Operators)
  - : Aprende más sobre el comportamiento de los operadores de JavaScript {{jsxref("Operators/instanceof", "instanceof")}}, {{jsxref("Operators/typeof", "typeof")}}, {{jsxref("Operators/new", "new")}}, {{jsxref("Operators/this", "this")}}, el [operador de precedencia](/es/docs/Web/JavaScript/Reference/Operators/Operator_Precedence ), y más.
- [Sentencias y declaraciones](/es/docs/Web/JavaScript/Reference/Statements)
  - : Aprende cómo funciona {{jsxref("Statements/do...while", "do-while")}}, {{jsxref("Statements/for...in", "for-in")}}, {{jsxref("Statements/for...of", "for-of")}}, {{jsxref("Statements/try...catch", "try-catch")}}, {{jsxref("Statements/let", "let")}}, {{jsxref("Statements/var", "var")}}, {{jsxref("Statements/const", "const")}}, {{jsxref("Statements/if...else", "if-else")}}, {{jsxref("Statements/switch", "switch")}}, y más declaraciones y palabras reservadas de JavaScript.
- [Funciones](/es/docs/Web/JavaScript/Reference/Functions)
  - : Aprende cómo trabajar con funciones de JavaScript para desarrollar tus aplicaciones.

## Herramientas &amp; recursos

> Herramientas útiles para escribir y depurar tu código **JavaScript**.

Reviewer 2: línea repetida

Reviewer: @Vallejoanderson aquí no veo el cambio, si lo hiciste?

Author: Sip, fue borrada :D

Herramientas útiles para escribir y depurar tu código **JavaScript**.

> - [Herramientas para Desarrolladores de Firefox](/es/docs/Tools)

Reviewer 2: Linea repetida

Author: Listo, borrada.

- [Herramientas para Desarrolladores de Firefox](/es/docs/Tools)
  - : [Consola Web](/es/docs/Tools/Web_Console), [Generador de Perfiles de JavaScript](/es/docs/Tools/Profiler), [Depurador](/es/docs/Tools/Debugger), y más.

> - [Javascripts Shells](/es/docs/Web/JavaScript/Shells)

Reviewer 2: Verificar `shell` (`consola`)

Author: Modificado.

  - : Un shell de JavaScript te permite probar rápidamente fragmentos de código JavaScript.
- [Aprende JavaScript](https://learnjavascript.online/)
  - : Un excelente recurso para aspirantes a desarrollador web — Aprendr JavaScript en un entorno interactivo, con lecciones cortas y pruebas interactivas, guiado por una evaluación automatizada. Las primeras 40 lecciones son gratis, y el curso completo está disponible por un pequeño pago único.
- [TogetherJS](https://togetherjs.com/)

>   - : La colaboración hecha fácil. Al agregar TogetherJS a tu sitio web, ¡tus usuarios pueden ayudarse mutuamente en un sitio web en tiempo real!

Reviewer 2: Revisar congruencia de la traducción

Author: Listo @Graywolf9, ya hice todos los cambios espero no se me haya pasado ninguno, me avisas si tienes otra observacion, por fa. Gracias por la ayuda.

---
title: JavaScript
slug: Web/JavaScript
tags:
 - Aprender
 - JavaScript
 - 'l10n:priority'
translation_of: Web/JavaScript
---
{{JsSidebar}}

> **JavaScript** (**JS**) es un lenguaje de programación ligero, interpretado, o compilado [justo-a-tiempo](https://en.wikipedia.org/wiki/Just-in-time_compilation) (just-in-time) con {{Glossary("First-class Function", "funciones de primera clase")}}. Si bien es más conocido como un lenguaje de scripting (secuencias de comandos) para páginas web, y es usado en [muchos entornos fuera del navegador](https://en.wikipedia.org/wiki/JavaScript#Uses_outside_Web_pages), tal como {{Glossary("Node.js")}}, [Apache CouchDB](https://couchdb.apache.org/) y [Adobe Acrobat](https://www.adobe.com/devnet/acrobat/javascript.html) JavaScript es un lenguaje de {{Glossary ("Prototype-based programming", "programación basada en prototipos")}}, multiparadigma, de un solo hilo, dinámico, con soporte para programación orientada a objetos, imperativa y declarativa (por ejemplo programación funcional). Lee más en [acerca de JavaScript](/es/docs/Web/JavaScript/About_JavaScript).

Reviewer 3: Some external URLs/resources in English also have translated versions. If you find that they're good quality, feel free to link to the Spanish version of that resource rather than the English one :)

I only checked a couple of the Wikipedia pages, but these look like they *could* be linked to in place of the English ones (this is, of course, completely up to you):

- https://es.wikipedia.org/wiki/Java_(lenguaje_de_programaci%C3%B3n)
- https://es.wikipedia.org/wiki/JavaScript

Reviewer: it totally makes sense, I also agree, I'll share this "small but minute detail" with the rest of the community. 

I'll add the suggestion for you, why don't have permission to suggest edits?

Reviewer 3: Oops, that was a mix-up on my part. I forgot the syntax for adding suggestions and for some reason thought that doing so required specific permissions, I'll edit them now :)

Reviewer: hey @cw118 taking advantage you are here, we were wondering if we could add an ISSUE_TEMPLATE for the Spanish version, it could be challenging for Spanisk Speakers to open a new issue and see the questions in English, we are thinking in maybe add the same questionary but translated to Spanish, what do you think?

Reviewer 3: I might not be the best person to ask (maybe a question for other maintainers/review teams?), but I personally think that'd be great :) 

And just to avoid filling up this PR/thread with lots of comments, would it be alright if I join Spanish MDN matrix chat? It'd be the perfect platform for discussions like these (and now I really have to learn Spanish)

Author: Hi @cw118. I am new at this and I appreciate your help. Thank you for helping us to have a more presentable PR (I hope I have expressed myself well) , I can't be more agree about your proposal to use links in Spanish instead of the English one.

Reviewer 3: **Just some important stuff to add to the external links detail**: 

- Some resources, including official specifications, are only available in English (e.g. `https://tc39.es/ecma402/`), so localizers just use the English resource
- Even with others where translations *are* available, they might not be "good enough quality" to include — this is up to the community (contributors and reviewers) to verify — in which case it could be better to stick with the original English link. 

I hope I explained that okay. Either way, if it's alright with everyone, I think I'm going to join the `l10n-es` Matrix chat and we could discuss everything further there?

> I am new at this and I appreciate your help. Thank you for helping us to have a more presentable PR (I hope I have expressed myself well)

Not a problem, it's so nice to see the Spanish community back in action here! You guys are doing great :)

> **JavaScript** (**JS**) es un lenguaje de programación ligero, interpretado, o compilado [justo-a-tiempo](https://en.wikipedia.org/wiki/Just-in-time_compilation) (just-in-time) con {{Glossary("First-class Function", "funciones de primera clase")}}. Si bien es más conocido como un lenguaje de scripting (secuencias de comandos) para páginas web, y es usado en [muchos entornos fuera del navegador](https://en.wikipedia.org/wiki/JavaScript#Uses_outside_Web_pages), tal como {{Glossary("Node.js")}}, [Apache CouchDB](https://couchdb.apache.org/) y [Adobe Acrobat](https://www.adobe.com/devnet/acrobat/javascript.html) JavaScript es un lenguaje de {{Glossary ("Prototype-based programming", "programación basada en prototipos")}}, multiparadigma, de un solo hilo, dinámico, con soporte para programación orientada a objetos, imperativa y declarativa (por ejemplo programación funcional). Lee más en [acerca de JavaScript](/es/docs/Web/JavaScript/About_JavaScript).

Reviewer: ```suggestion
**JavaScript** (**JS**) es un lenguaje de programación ligero, interpretado, o compilado [justo-a-tiempo](https://es.wikipedia.org/wiki/Compilaci%C3%B3n_en_tiempo_de_ejecuci%C3%B3n) (just-in-time) con {{Glossary("First-class Function", "funciones de primera clase")}}. Si bien es más conocido como un lenguaje de scripting (secuencias de comandos) para páginas web, y es usado en [muchos entornos fuera del navegador](https://es.wikipedia.org/wiki/JavaScript), tal como {{Glossary("Node.js")}}, [Apache CouchDB](https://couchdb.apache.org/) y [Adobe Acrobat](https://www.adobe.com/devnet/acrobat/javascript.html) JavaScript es un lenguaje de {{Glossary ("Prototype-based programming", "programación basada en prototipos")}}, multiparadigma, de un solo hilo, dinámico, con soporte para programación orientada a objetos, imperativa y declarativa (por ejemplo programación funcional). Lee más en [acerca de JavaScript](/es/docs/Web/JavaScript/About_JavaScript).
```

Esta sección está dedicada al lenguaje JavaScript en sí, y no a las partes que son específicas de las páginas web u otros entornos host. Para información acerca de {{Glossary("API","APIs")}} específicas para páginas Web, consulta [APIs Web](/es/docs/Web/API ) y {{Glossary("DOM")}}.

El estándar para JavaScript es [ECMAScript](/es/docs/Web/JavaScript/Language_Resources) (ECMA-262) y [la especificación de la API para la Internacionalización de ECMAScript](https://tc39.es/ecma402/) (ECMA-402). La documentación en MDN está basada enteramente en las últimas versiones preliminares de ECMA-262 y ECMA-402. Y en algunos casos donde algunas [propuestas para nuevas funciones para ECMAScript](https://github.com/tc39/proposals) ya hayan sido implementadas en los navegadores, la documentación y algunos artículos de MDN pueden hacer uso de algunas de estas funciones.

> No confundas JavaScript con el [lenguaje de programación Java](https://en.wikipedia.org/wiki/Java_(programming_language) ). Ambos "Java" y "JavaScript" son marcas o marcas registradas de Oracle en los Estados Unidos y otros países, Sin embargo, los dos lenguajes de programación tienen sintaxis, semántica y usos muy diferentes.

Reviewer 3: Looks like this `(https://en.wikipedia.org/wiki/Java_(programming_language) )` has an extra space and closing parenthesis `)` at the end, in addition to the possibility of linking to the Spanish counterpart and not the English one (see comment above)

Reviewer: No, it is not an extra `)` it is part of the wiki URL, https://es.wikipedia.org/wiki/Java_(lenguaje_de_programaci%C3%B3n)

Reviewer 3: Got it, didn't see that 👍

> > **Callout:** **Quieres transformarte en un desarrollador web front-end?**

Reviewer 3: Once the new card keywords are added for the `es` locale (see [this pull request by JuanVqz at mdn/yari](https://github.com/mdn/yari/pull/5225), the **Callout:** will probably have to be changed to the **Observación:** keyword

Reviewer 3: N.B. if this is changed to the Spanish keyword before the yari PR is merged, it won't have the expected styling, but that problem should "go away" once the `es.json` file is created

Reviewer: good to know because I thought yari will find the word `Callout` and replace it by itself.

> > **Callout:** **Quieres transformarte en un desarrollador web front-end?**

Reviewer: Yo habia pensado que `Observación` era una buena traducción para `Callout` pero en este parrafo lo estoy dudando, que piensan ustedes? @mdn/yari-content-es

```suggestion
> **Observación:** **Quieres transformarte en un desarrollador web front-end?**
```

Author: Suena bien, aunque me preocupa que no se estilice como sucede en la version en ingles.

>
> Hemos reunido un curso que incluye toda la información esencial que necesitas para trabajar hacia tu objetivo.
> 
> [Empieza aquí](es/docs/Learn/Front-end_web_developer)
> 

## Tutoriales

Aprende a programar en JavaScript con guías y tutoriales.

### Para completos principiantes

> Dirígete a nuestro tema del [Area de Aprendizaje JavaScript](/es/docs/Learn/JavaScript ) si quieres aprender JavaScript pero no tienes experiencia previa de JavaScript o programación. Los módulos completos disponibles allí son los siguientes:

Reviewer 3: Being a bit picky here, but there's an extra space after `(/es/docs/Learn/JavaScript )` (in the link)

Reviewer: good eye! :+1:

> Dirígete a nuestro tema del [Area de Aprendizaje JavaScript](/es/docs/Learn/JavaScript ) si quieres aprender JavaScript pero no tienes experiencia previa de JavaScript o programación. Los módulos completos disponibles allí son los siguientes:

Reviewer: ```suggestion
Dirígete a nuestro tema del [Area de Aprendizaje JavaScript](/es/docs/Learn/JavaScript) si quieres aprender JavaScript pero no tienes experiencia previa de JavaScript o programación. Los módulos completos disponibles allí son los siguientes:
```

Reviewer 3: Sorry, I started converting my comments to suggestions — feel free to delete/resolve my old ones or ask me to delete mine :)

  - : Explora qué son las APIs y cómo usar algunas de las APIs más comunes que encontrarás a menudo en tu trabajo de desarrollo.

### Guía de JavaScript

- [Guía de JavaScript](/es/docs/Web/JavaScript/Guide)
  - : Una guía mucho más detallada del lenguaje JavaScript, dirigida a aquellos con experiencia previa en programación, ya sea en JavaScript u otro lenguaje.

### Intermedio

- [Comprendiendo los frameworks JavaScript de lado del cliente](/es/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks)
  - : Los frameworks de JavaScript son una parte esencial del desarrollo web front-end moderno, ya que proporcionan a los desarrolladores herramientas probadas y comprobadas para crear aplicaciones web interactivas y escalables. Este módulo te brinda algunos conocimientos básicos sobre cómo funcionan los frameworks del lado del cliente y cómo encajan en tu conjunto de herramientas, antes de pasar a la serie de tutoriales que cubren algunos de los más populares en la actualidad.
- [Una reintroducción a JavaScript](/es/docs/Web/JavaScript/A_re-introduction_to_JavaScript)
  - : Una descripción general para aquellos que _piensan_ que saben JavaScript.

> 

Reviewer 3: Unnecessary newline

```suggestion
- [Una reintroducción a JavaScript](/es/docs/Web/JavaScript/A_re-introduction_to_JavaScript)
  - : Una descripción general para aquellos que _piensan_ que saben JavaScript.
- [Estructuras de datos en JavaScript](/es/docs/Web/JavaScript/Data_structures)
  - : Descripción general de las estructuras de datos disponibles en JavaScript.
```

> 

Reviewer: ```suggestion
```

  - : Un cierre es la combinación de una función y un entorno léxico dentro del cual se declaró esa función.

### Avanzado

- [Herencia y la cadena de prototipos](/es/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
  - : Explicación de la herencia basada en prototipos ampliamente incomprendida y subestimada.
- [Modo estricto](/es/docs/Web/JavaScript/Reference/Strict_mode)
  - : El modo estricto define que no puedes usar ninguna variable antes de iniciarla. Es una variable restringida de ECMAScript 5, para un rendimiento más rápido y una depuración más fácil.
- [Arreglos escritos en JavaScript](/es/docs/Web/JavaScript/Typed_arrays)
  - : Los arreglos escritos en JavaScript proveen un mecanismo para acceder a datos binarios sin formato.
- [Gestión de la Memoria](/es/docs/Web/JavaScript/Memory_Management)
  - : Ciclo de vida de la memoria y recolección de basura en JavaScript.
- [Modelo de Concurrencia y Bucle de Eventos](/es/docs/Web/JavaScript/EventLoop)
  - : JavaScript tiene un modelo de concurrencia basado en un "bucle de eventos".

## Referencia

Consulta la documentación completa de la [referencia de JavaScript](/es/docs/Web/JavaScript/Reference).

> 

Reviewer 3: Being a little picky, but there's an extra newline here

Reviewer: I have seen it but I decided not to say anything since now we are two, I think it's time to do it.

Do you know a good `Prettier`, `linter`, or `auto-formatted for markdown? which editor?

Reviewer 3: There are lots out there, I've recently been trying out [`markdownlint`](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) in VS Code

> 

Reviewer: ```suggestion
```

- [Expresiones y operadores](/es/docs/Web/JavaScript/Reference/Operators)
  - : Aprende más sobre el comportamiento de los operadores de JavaScript {{jsxref("Operators/instanceof", "instanceof")}}, {{jsxref("Operators/typeof", "typeof")}}, {{jsxref("Operators/new", "new")}}, {{jsxref("Operators/this", "this")}}, el [operador de precedencia](/es/docs/Web/JavaScript/Reference/Operators/Operator_Precedence ), y más.
- [Sentencias y declaraciones](/es/docs/Web/JavaScript/Reference/Statements)
  - : Aprende cómo funciona {{jsxref("Statements/do...while", "do-while")}}, {{jsxref("Statements/for...in", "for-in")}}, {{jsxref("Statements/for...of", "for-of")}}, {{jsxref("Statements/try...catch", "try-catch")}}, {{jsxref("Statements/let", "let")}}, {{jsxref("Statements/var", "var")}}, {{jsxref("Statements/const", "const")}}, {{jsxref("Statements/if...else", "if-else")}}, {{jsxref("Statements/switch", "switch")}}, y más declaraciones y palabras reservadas de JavaScript.
- [Funciones](/es/docs/Web/JavaScript/Reference/Functions)
  - : Aprende cómo trabajar con funciones de JavaScript para desarrollar tus aplicaciones.

> 

Reviewer 3: Another extra newline here **and** at line 96 (or 97, depending on which one you decide to remove)

Reviewer 3: N.B. There are some "extra" single newlines between bullet points as well (e.g. line 57), for example:

- Text here

- More text here

I didn't comment on those occurrences since they technically shouldn't change the formatting/layout of the page in the end, so it really has to do with Markdown syntax/style and consistency

Reviewer: @cw118 if you still helping us, you will learn Spanish :smile:

> 

Reviewer: ```suggestion
```

## Herramientas &amp; recursos

Herramientas útiles para escribir y depurar tu código **JavaScript**.

> 

Reviewer 3: ```suggestion
Herramientas útiles para escribir y depurar tu código **JavaScript**.

- [Herramientas para Desarrolladores de Firefox](/es/docs/Tools)
```

- [Aprende JavaScript](https://learnjavascript.online/)
  - : Un excelente recurso para aspirantes a desarrollador web — Aprendr JavaScript en un entorno interactivo, con lecciones cortas y pruebas interactivas, guiado por una evaluación automatizada. Las primeras 40 lecciones son gratis, y el curso completo está disponible por un pequeño pago único.
- [TogetherJS](https://togetherjs.com/)
  - : La colaboración hecha fácil. Al agregar TogetherJS a tu sitio web, ¡tus usuarios pueden ayudarse mutuamente en un sitio web en tiempo real!
- [Stack Overflow](https://stackoverflow.com/questions/tagged/javascript)
  - : Preguntas en Stack Overflow etiquetadas con "JavaScript".
- [JSFiddle](https://jsfiddle.net/)
  - : Edita JavaScript, CSS, HTML y obtén resultados en vivo. Usa recursos externos y colabora con tu equipo en línea.
- [Plunker](https://plnkr.co/)
  - : Plunker es una comunidad en línea para crear, colaborar y compartir sus ideas de desarrollo web. Edita sus archivos JavaScript, CSS, HTML y obtén resultados en vivo y estructuras de archivos.
- [JSBin](https://jsbin.com/)
  - : JSBin es una herramienta de depuración de desarrollo web colaborativo de código abierto.

>  

Reviewer 3: ```suggestion
- [JSBin](https://jsbin.com/)
  - : JSBin es una herramienta de depuración de desarrollo web colaborativo de código abierto.
- [Codepen](https://codepen.io/) 
  - : Codepen es otra herramienta colaborativa de desarrollo web usada como un espacio para mostrar resultados en vivo.
```

>  

Reviewer: ```suggestion
```

- [Codepen](https://codepen.io/) 
  - : Codepen es otra herramienta colaborativa de desarrollo web usada como un espacio para mostrar resultados en vivo.

>  

Reviewer 3: ```suggestion
- : Codepen es otra herramienta colaborativa de desarrollo web usada como un espacio para mostrar resultados en vivo.
- [StackBlitz](https://stackblitz.com/)
  - : StackBlitz es otra herramienta de depuración/visualización en línea, que puede alojar e implementar aplicaciones full-stack usando React, Angular, etc.
```

>  

Reviewer: ```suggestion
```
